docker/docker_client.go

```diff
@@ -187,27 +187,31 @@ func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool)
 	}, nil
 }
 
+type requestOptions struct {
+	remoteName string
+	actions    string
+}
```

Comment:
on a second though, `makeRequest` and `makeRequestToResolvedURL` already take too many param IMO. I would rather compact those options (and others) in a generic `requestOptions` struct. Mandatory arguments continue to be passed as function arguments though.